### PR TITLE
Default to Latest SnapRAID Release & Use Debian Bullseye

### PR DESCRIPTION
**Pull request by Kmagameguy**
Hello there 👋️

I am a big fan of the self-hosted.show podcast and have been using a lot of the suggested tools & guidance to grow my homelab (I'm KmaGameGoof on the Discord).  This docker repo has been awesome for building SnapRAID updates since it saves me from cluttering my main server with build tools.

### What's Changed
I figured I would try to give back a bit by proposing this PR.  It does the following:
1. Queries the Github API for the latest SnapRAID release tag for use in the build.  As of time of writing, this is `12.0`.
    - The Github API allows [60 unauthenticated requests per hour](https://docs.github.com/en/rest/overview/resources-in-the-rest-api#rate-limiting), which I think is suitable for consumers of this repo.
    - As  a fallback, the changes preserve the ability to call the script with an explicit release tag, e.g. `11.2`.
2. Pushes the Dockerfile up to Debian Bullseye, since that's the latest `stable` release of Debian
3. Updates a few minor details in the README

### Reasoning for Proposed Changes
Admittedly, all of this could be considered unnecessary, since you can pass a release version to the build script.  My interest in proposing these changes are as follows:
1. For someone utterly new to self-hosting, they won't have to research the SnapRAID repo to check for the latest release
2. It could be used in a cronjob to set & forget SnapRAID updates (I plan to do precisely this)
3. I believe I saw on the Discord that you might stop using SnapRAID in future media server builds; if that's the case, this PR makes it so you don't really have to maintain the repo anymore. I know SnapRAID isn't updated often but hopefully it means you won't have to attend to PRs/Twitter DMs/Discord questions about new releases, since the default install will always pull the latest

### Other Notes
I have already produced a `snapraid.deb` artifact on my homelab using this PR, and I've also run it against the provided CI configuration on my own repo, for whatever that may be worth (hopefully this link isn't set to private):
https://github.com/Kmagameguy/docker-snapraid/actions/runs/1599930507

Lastly, thank you for sharing your knowledge & experiences with the 'net!

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,10 +1,10 @@
-FROM debian:buster
+FROM debian:bullseye
 MAINTAINER Alex Kretzschmar <alexktz@gmail.com>
 
-ARG SNAPRAID_VERSION="11.6"
+ARG SNAPRAID_VERSION
 
 # Builds SnapRAID from source
-RUN echo 'deb http://deb.debian.org/debian buster-backports main' > /etc/apt/sources.list.d/backports.list && \
+RUN echo 'deb http://deb.debian.org/debian bullseye-backports main' > /etc/apt/sources.list.d/backports.list && \
```
Comment (RealOrangeOne) [inline Dockerfile:7]: I'd be interested to know whether backports is necessary anymore?

File: Dockerfile [hunk 2]
```diff
@@ -1,10 +1,10 @@
-FROM debian:buster
+FROM debian:bullseye
```
Comment (RealOrangeOne) [inline Dockerfile:1]: Maybe it'll work if it's unpinned?

```suggestion
FROM debian
```

Comment (Kmagameguy) [reply Dockerfile]: Good thinking, works like a charm.

File: Dockerfile (hunk 1 above)
Comment (Kmagameguy) [reply Dockerfile]: Seems to work just fine without it now 👍🏼

**Comment by Kmagameguy** [comment]
@RealOrangeOne Thank you for taking the time to review this PR and for providing some comments.  I made two changes based on your notes:
1. Unpinned the Debian version; seemed to pull bullseye just fine in my local dev / forked repo
2. I also removed the backports source; again, seemed to build just fine in my local dev / forked repo

I've made these changes inside two separate commits so the history is easier to traverse.  Happy to rebase for a cleaner commit history if this PR seems worth pursuing overall.

**Review by ironicbadger — APPROVED** [review]

**Comment by ironicbadger** [comment]
Very nice idea, thanks for the PR!

**Pull request merged**